txscript/error.go

```diff
@@ -231,10 +231,11 @@ const (
 	// the witness stack itself is empty.
 	ErrWitnessProgramEmpty
 
-	// ErrWitnessScriptMismatch is returned if ScriptVerifyWitness is set
+	// ErrWitnessProgramMismatch is returned if ScriptVerifyWitness is set
 	// and the witness itself for a p2wkh witness program isn't *exactly* 2
-	// items.
-	ErrWitnessScriptMismatch
+	// items or if the witness for a p2wsh isn't the sha255 of the witness
+	// script.
+	ErrWitnessProgramMismatch
```

Comment:
I think most of the segwit errors, perhaps with the exception of `ErrDiscourageUpgradableWitnessProgram`, `ErrMinimalIf`, `ErrWitnessMalleated` and `ErrWitnessMalleatedP2SH` should be at the bottom under a new heading:

```Go
 	// ----------------------------------------
 	// Failures related to segregated witness.
	// ----------------------------------------
```

The rationale is that `ErrDiscourageUpgradableWitnessProgram` is clearly related to soft forking (specifically the safety thereof), so should go under the soft forks heading, and `ErrMinimalIf`, `ErrWitnessMalleated` and `ErrWitnessMalleatedP2SH` are related to malleability, so should go under the malleability section, but everything else only applies to segwit programs.

Finally, whether you move them or not, please make sure the order in the `errorCodeStrings` map and `TestErrorCodeStringer` tests are all defined in the same order for consistency.